### ISSUE TITLE
Way to handle disrupted connection

### ISSUE THREAD

**Issue by volbil** [aiidateam/kiwipy#105]
Hello, first of all thanks for this awesome library! I was wondering if there is native/good way to handle RabbitMQ disrupted connection and retry connection if it failed?

**Comment by muhrin** [comment]
Hi @volbil , thanks for the kind words.

Could you give a few more details of what you have in mind?

`kiwipy` uses aio_pika's [connect_robust](https://aio-pika.readthedocs.io/en/latest/apidoc.html?highlight=robust#aio_pika.connect_robust) method under the hood which will make an attempt to reconnect (and recreate channels/queues, etc) if it detects that the connection is dropped.

I'm not sure how long for/how many times `connect_robust` will attempt to reconnect after a connection drop though.  Perhaps it gives up after some time.

Do you have an exception from a situation like this you could share?

**Comment by volbil** [comment]
Sure, it's preatty easy to replicate. I used example code for rpc server https://github.com/aiidateam/kiwipy/blob/develop/examples/rmq_rpc_server.py and when it was running, I stopped RabbitMQ to see what would happen. First message was `Connection attempt to "amqp://127.0.0.1" failed. Reconnecting after 5 seconds.` and afther that `asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of 1 expected bytes` error happened (full error log can be seen here https://paste.ubuntu.com/p/5CfBg73qPM/).

Once I turned RabbitMQ back error messages stopped appearing and kiwipy started working as intended (which is good). I know I can handle `IncompleteReadError` exception using try/except but maybe it should be handled on kiwipy side?
